tests/test_sparse_matrix_device_operator.cu

```diff
@@ -0,0 +1,145 @@
+/*************************************************************************
+ * Copyright (c) 2018 by the mfmg authors                                *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#define BOOST_TEST_MODULE sparse_matrix_device_operator
+
+#include "main.cc"
+
+#include <mfmg/dealii_operator_device.cuh>
+#include <mfmg/utils.cuh>
+
+#include <set>
+#include <utility>
+
+BOOST_AUTO_TEST_CASE(matrix_operator)
+{
+  // Create the cusparse_handle
+  cusparseHandle_t cusparse_handle = nullptr;
+  cusparseStatus_t cusparse_error_code;
+  cusparse_error_code = cusparseCreate(&cusparse_handle);
+  mfmg::ASSERT_CUSPARSE(cusparse_error_code);
+
+  unsigned int const n_rows = 30;
+  unsigned int const nnz_per_row = 1;
```

Comment:
Probably want to change it to something greater than 1. Otherwise, `n_rows` = `n_cols`, and the checks for `domain` and `range` can hide a problem.

Reply:
Yeah I had a bug that I was trying to fix and I forgot to change the value back.